### PR TITLE
{bio}[foss/2023a] AlphaPulldown v2.0.0 Uni-Core v0.0.3 w/ CUDA v12.1.1

### PR DESCRIPTION
**Pull request by ThomasHoffmann77**
(created using `eb --new-pr`)
requires:
- [x] #20362 

adapted from PR #21179

### CODE REVIEW

**Comment by boegel** [comment]
@ThomasHoffmann77 Additional extensions are required for the `convert_to_modelcif.py` script, see 99abe91b43 (included in #21179)

**Comment by ThomasHoffmann77** [comment]
switched to draft: expecting 2.0.0b6 with setup.cfg not using hardcoded lib/python3.10/site-packages (see https://github.com/KosinskiLab/AlphaPulldown/commit/cc4b0af60518c078305bbe4c584691d1ed9ade31)

File: test/easyconfigs/easyconfigs.py [hunk 1]
```diff
@@ -655,6 +655,8 @@ def check_dep_vars(self, gen, dep, dep_vars):
                 ('2.5.3;', ['medaka-1.5.0-']),
                 # tensorflow-probability version to TF version
                 ('2.8.4;', ['tensorflow-probability-0.16.0-']),
+                # AlphaPulldown
+                ('2.15.1', ['AlphaPulldown-2.0.0b6']),
```
Comment (akesandgren) [inline test/easyconfigs/easyconfigs.py:659]: This shouldn't be required any longer right?

Comment (ThomasHoffmann77) [reply test/easyconfigs/easyconfigs.py]: @akesandgren I removed the line
@jkosinski

File: easybuild/easyconfigs/a/AlphaPulldown/AlphaPulldown-2.0.0-foss-2023a-CUDA-12.1.1.eb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    ('CUDA', '12.1.1', '', SYSTEM),
+    ('Python', '3.11.3'),
+    ('OpenMM', '8.0.0', versionsuffix),
+    ('Kalign', '3.4.0'),
+    ('PyYAML', '6.0'),
+    ('jax', '0.4.25', versionsuffix),  # also provides absl-py
+    ('Biopython', '1.83'),
+    ('h5py', '3.9.0'),
+    ('IPython', '8.14.0'),
+    ('JupyterLab', '4.0.5'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/AlphaPulldown/AlphaPulldown-2.0.0-foss-2023a-CUDA-12.1.1.eb:29]: JupyterLab needs to be last.
jupyter-events from jupyter-server (which is a dep in JupyterLab) has a dependency for jsonschema >= 4.18.0 but everything else in this dep list uses 4.17.3.

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/AlphaPulldown/AlphaPulldown-2.0.0-foss-2023a-CUDA-12.1.1.eb (hunk 2 above)
Comment (ThomasHoffmann77) [reply easybuild/easyconfigs/a/AlphaPulldown/AlphaPulldown-2.0.0-foss-2023a-CUDA-12.1.1.eb]: @akesandgren thanks. I moved it.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
b-cn1611.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 555.58.02, Python 3.10.12
See https://gist.github.com/akesandgren/56aea3df4b4f9d09c1011edb6e7d97b3 for a full test report.

File: easybuild/easyconfigs/a/AlphaPulldown/AlphaPulldown-2.0.0-foss-2023a-CUDA-12.1.1.eb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        'sources': [{
+            'filename': '%(name)s-%(version)s.tar.gz',
+            'git_config': {
+                'url': 'https://github.com/KosinskiLab',
+                'repo_name': 'AlphaPulldown',
+                'tag': version, 'recursive': True
+            }
+        }],
+        'testinstall': True,
+        'checksums': ['e338195987e003f3caadb06bda0ca56dece87e358738143ea72662f9ad69b1d4'],
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/AlphaPulldown/AlphaPulldown-2.0.0-foss-2023a-CUDA-12.1.1.eb:101]: This needs to be [None], at least until EB v5 is out

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/AlphaPulldown/AlphaPulldown-2.0.0-foss-2023a-CUDA-12.1.1.eb (hunk 3 above)
Comment (ThomasHoffmann77) [reply easybuild/easyconfigs/a/AlphaPulldown/AlphaPulldown-2.0.0-foss-2023a-CUDA-12.1.1.eb]: @akesandgren done

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
b-cn1611.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 555.58.02, Python 3.10.12
See https://gist.github.com/akesandgren/48ee3cf53275a000291a44b6c8a8e5e2 for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @ThomasHoffmann77!

**Pull request merged**